File: baseshemas4-1.lua

local shemat41 = function(marche) 
    for i = 1,5  do
        turtle.forward(marche)
        turtle.backward(marche//2)
        turtle.turnLeft(90)
    end
end

local goBottomToLeft = function(marche)
    turtle.turnRight(90)
    turtle.forward(marche//2)
    turtle.turnRight(180)
    turtle.forward(marche)
    turtle.turnLeft(90)
    turtle.forward(marche//2)
end



local goLeftToTop = function(marche)
    turtle.turnRight(180)
    turtle.forward(marche)
    turtle.turnLeft(90)
    turtle.forward(marche//2)
end


local goTopToRight = function(marche)

    turtle.turnRight(180)
    turtle.forward(marche)
    turtle.turnLeft(90)
    turtle.forward(marche//2)

end



local goRightToBottom = function(marche)
    turtle.turnRight(180)
    turtle.forward(marche)
    turtle.turnLeft(90)
    turtle.forward(marche//2)
end



shemat41(100)

for i=1,100 do
    choix = math.random(1, 4)

    if choix == 1 then
        goBottomToLeft(100)
    elseif choix == 2 then
        goBottomToLeft(100)
        goLeftToTop(100)
    elseif choix == 3 then
        goBottomToLeft(100)
        goLeftToTop(100)
        goTopToRight(100)
    elseif choix == 4 then
        goBottomToLeft(100)
        goLeftToTop(100)
        goTopToRight(100)
        goRightToBottom(100)
    end

end

File: baseshemas4-1.py
from random import randint
import turtle

turtle.bgcolor("black")

x=0
y= 0
turtle.goto(x, y)



def shemat41(color, marche):

    turtle.color(color)
    for i in range(1, 5):

        turtle.forward(marche)
        turtle.backward(marche//2)
        turtle.left(90)


def goBottomToLeft(marche):
    turtle.right(90)
    turtle.forward(marche//2)
    turtle.right(180)
    turtle.forward(marche)
    turtle.left(90)
    turtle.forward(marche//2)


def goLeftToTop(marche):
    turtle.right(180)
    turtle.forward(marche)
    turtle.left(90)
    turtle.forward(marche//2)


def goTopToRight(marche):
    turtle.right(180)
    turtle.forward(marche)
    turtle.left(90)
    turtle.forward(marche//2)


def goRightToBottom(marche):
    turtle.right(180)
    turtle.forward(marche)
    turtle.left(90)
    turtle.forward(marche//2)




shemat41("red", 100)

for i in range(1,100):
   
    choix = randint(1,4-1)


    if choix == 1:
        goBottomToLeft(100)


    if choix == 2:
        goBottomToLeft(100)
        goLeftToTop(100)


    if choix == 3:
        goBottomToLeft(100)
        goLeftToTop(100)
        goTopToRight(100)
 

    if choix == 4 :
        goBottomToLeft(100)
        goLeftToTop(100)
        goTopToRight(100)
        goRightToBottom(100)
  
input("Appuyer une touche pour arreter ........ ")





